driver/src/orderbook/streamed/balance.rs

```diff
@@ -98,24 +98,34 @@ impl Balance {
     }
 
     pub fn sell(&mut self, amount: u128, batch_id: BatchId) -> Result<()> {
-        self.sell_buy_internal(amount, batch_id, TradeType::Sell)
+        self.sell_buy_internal(amount.into(), batch_id, TradeType::Sell)
     }
 
     pub fn buy(&mut self, amount: u128, batch_id: BatchId) -> Result<()> {
-        self.sell_buy_internal(amount, batch_id, TradeType::Buy)
+        self.sell_buy_internal(amount.into(), batch_id, TradeType::Buy)
     }
 
     pub fn revert_sell(&mut self, amount: u128, batch_id: BatchId) -> Result<()> {
-        self.revert_sell_buy_internal(amount, batch_id, TradeType::Sell)
+        self.revert_sell_buy_internal(amount.into(), batch_id, TradeType::Sell)
     }
 
     pub fn revert_buy(&mut self, amount: u128, batch_id: BatchId) -> Result<()> {
-        self.revert_sell_buy_internal(amount, batch_id, TradeType::Buy)
+        self.revert_sell_buy_internal(amount.into(), batch_id, TradeType::Buy)
+    }
+
+    pub fn solution_submission(&mut self, fee: U256, batch_id: BatchId) -> Result<()> {
```

Comment:
Is there a benefit of having different methods for this if its logical equivalent to a buy/sell? Maybe we can find a name like credit/debit that capture the act of increasing/decreasing balance better.

Reply:
As I see it doing it this way makes `State` a little simpler, where it becomes responsible for finding the `Balance`s and `Order`s that need to be modified and then forwarding the needed event data to those components. I think its nice that each event that can change a `Balance` has their own method which completely encapsulates all the required logic for that event.

It also allows you to look at `Balance` in terms of a stream of events. For example you can, independent of the logic in `State` create a unit test that applies these events in a sequence.

Also, from a quick glance at the code, `Trade`s and `TradeReversion`s have slightly different semantics with respect to applying the pending fluxes and proceeds so that information would be in `State` instead of `Balance`s.

Reply:
I did it for the reasons Nick outlined. From state's perspective I find it a bit easier to have this method and it shows that you could change the implementation of it in balance without breaking anything.